runtime/component/dnd-mode.js

```diff
@@ -0,0 +1,634 @@
+  CheckSwitchOff: 2,
+  CheckStatusOff: 3,
+  CheckStatusOn: 4,
+  DNDModeDisabled: 5,
+  CheckStatusOnX: 6,
+  CheckStatusOffX: 7,
+  CheckTimeSuccess: 8,
+  CheckTimeFailed: 9,
+  DNDModeDisabledX: 10,
+  CheckTimeSuccessX: 11,
+  CheckTimeFailedX: 12,
+  CheckActivityTrue: 13,
+  CheckActivityFalse: 14,
+  CheckActivityTrueX: 15,
+  CheckActivityFalseX: 16,
+  DNDModeEnabled: 17,
+  CheckAgain: 18
+}
+
+var SWITCH_KEY = 'dndmode.switch'
+var STATUS_KEY = 'dndmode.status'
+var WEEK_KEY = 'dndmode.weeksetting'
+var START_TIME_KEY = 'dndmode.starttime'
+var END_TIME_KEY = 'dndmode.endtime'
+var AWAKE_SWITCH_KEY = 'dndmode.awakeswitch'
+
+var FSM_READY = 0
+var FSM_RUNNING = 1
+var FSM_WAITING = 2
+var FSM_END = 3
+var FSM_ERROR = 4
+
+class DNDCommon {
+  constructor (life) {
+    this.life = life
+  }
+  /**
+   * Disable dnd mode
+   * @function disable
+   */
+  disable () {
+    // TODO disable dnd-mode
+    logger.info('dnd mode turned off')
+  }
+
+  /**
+   * Enable dnd mode
+   * @function enable
+   */
+  enable () {
+    // TODO enable dnd-mode
+    logger.info('dnd mode turned on')
+  }
+
+  /**
+   * Get dnd mode time span from now to start/end, return positive number if in dnd mode time
+   * @function getDNDTime
+   * @returns {number} - if result >= 0, it's the millisecond to end time
+   *                   - if result < 0, it's the millisecond to start time
+   */
+  static getDNDTime () {
+    function formatDate (dt) {
+      return `${dt.getFullYear()}-${dt.getMonth() + 1}-${dt.getDate()} ${dt.getHours()}:${dt.getMinutes()}:${dt.getSeconds()}`
+    }
+    var now = new Date()
+    var weekSetting = DNDCommon.getWeekSetting()
+    var start = DNDCommon.formatTime(DNDCommon.getStartTime(), 22, 0) // default start time is '22:00'
+    var end = DNDCommon.formatTime(DNDCommon.getEndTime(), 7, 0) // default end time is '07:00'
+    var weekDayStart = start.getDay()
+    var loopCount = 0
+    while (!weekSetting[weekDayStart % 7]) {
+      weekDayStart++
+      loopCount++
+      if (loopCount > 6) {
+        logger.info('all days of week is disabled, never been triggered')
+        return Number.MAX_VALUE
+      } else {
+        start.setDate(start.getDate() + 1)
+        end.setDate(end.getDate() + 1)
+      }
+    }
+    if (start >= end) {
+      end.setDate(end.getDate() + 1)
+    }
+    logger.info(`check local time(${now.getTimezoneOffset() / 60}) now:${formatDate(now)} \
+      start:${formatDate(start)}(week day:${start.getDay()}) end:${formatDate(end)}`)
+    if (now >= start && now < end) {
+      return end - now
+    } else if (now < start) {
+      return now - start
+    } else {
+      throw (new Error(`the dnd-mode end time(${formatDate(end)}) should always be larger \
+        than now (${formatDate(now)})`))
+    }
+  }
+
+  /**
+   * Get the switch value
+   * @function getSwitch
+   * @returns {string} switch value
+   */
+  static getSwitch () {
+    return property.get(SWITCH_KEY, 'persist')
```

Comment:
Do we really need this alias function? Because just found other KEY you were using without declaring alias :)